lib/src/test/java/uk/gov/di/ipv/cri/address/library/service/AddressServiceTest.java

```diff
@@ -143,4 +151,330 @@ void shouldGetAddressItem() {
         addressService.getAddressItem(SESSION_ID);
         verify(mockDataStore).getItem(String.valueOf(SESSION_ID));
     }
+
+    @Test
+    void testAddressesLinkerNoAddressesPasses() {
+        List<CanonicalAddress> canonicalAddresses = new ArrayList<>();
+        assertDoesNotThrow(() -> addressService.setAddressValidity(canonicalAddresses));
+    }
+
+    @Test
+    void testLinkerIsCurrentAddressWithNULLNULLEvaluatesToPrevious() {
+        // The reason for the linker existing is that a second address, intended as
+        // a PREVIOUS address has ValidFrom and ValidUntil not set - null,null. This however
+        // matches the pattern for a CURRENT address with unknown start date.
+        // In this specific case, the Linker evaluates null, null as the intended PREVIOUS address.
+
+        CanonicalAddress currentAddress = new CanonicalAddress();
+        currentAddress.setValidFrom(null);
+        currentAddress.setValidUntil(null);
+
+        List<CanonicalAddress> canonicalAddresses = List.of(currentAddress);
+
+        assertFalse(addressService.isCurrentAddress(currentAddress));
+        assertTrue(addressService.isNotCurrentAddress(currentAddress));
+
+        AddressProcessingException exception =
+                assertThrows(
+                        AddressProcessingException.class,
+                        () -> addressService.setAddressValidity(canonicalAddresses));
+
+        assertFalse(addressService.isCurrentAddress(currentAddress));
+        assertTrue(addressService.isNotCurrentAddress(currentAddress));
+
+        assertEquals(AddressService.ERROR_SINGLE_ADDRESS_NOT_CURRENT, exception.getMessage());
+    }
+
+    @Test
+    void testLinkerFailsToProcessWhenItWouldTrampleAlreadySetDates() {
+        // When AddressCRIFront is setting ValidUntil in previous addresses
+        // The linker can be removed entirely.
+
+        final LocalDate TEST_DATE_0 = LocalDate.of(1999, 12, 31);
+        // New date with one month gap as the linking if performed would use the first date
+        final LocalDate TEST_DATE_1 = LocalDate.of(1999, 11, 30);
+
+        CanonicalAddress currentAddress = new CanonicalAddress();
+        currentAddress.setValidFrom(TEST_DATE_0);
+        currentAddress.setValidUntil(null);
+
+        CanonicalAddress previousAddress = new CanonicalAddress();
+        previousAddress.setValidFrom(null);
+        previousAddress.setValidUntil(TEST_DATE_1);
+
+        List<CanonicalAddress> canonicalAddresses = List.of(currentAddress, previousAddress);
+
+        assertTrue(addressService.isCurrentAddress(currentAddress));
+        assertTrue(addressService.isNotCurrentAddress(previousAddress));
+
+        AddressProcessingException exception =
```

Comment:
add the error message as the final parameter to assertThrows and remove the assertEquals below
Please update every unit test method where applicable